lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2041,6 +2041,101 @@ namespace {
       fd->getAttrs().add(new (Impl.SwiftContext) UnsafeAttr(/*Implicit=*/true));
     }
 
+    void
+    generateStaticFactoryForCXXForeignRef(clang::ASTContext &clangCtx,
+                                          ClassDecl *result,
+                                          clang::CXXRecordDecl *cxxRecordDecl,
+                                          ClangImporter::Implementation &Impl) {
+
+      // Creating a name for the new synthesized function
+      clang::IdentifierTable &idents = clangCtx.Idents;
+      clang::IdentifierInfo *generatedFuncName =
+          &idents.get(("__returns" + cxxRecordDecl->getNameAsString()).c_str());
+
+      // Creating the return type for the synthesized function to be the
+      // pointe to cxxRecordDecl
+      const clang::Type *cxxRecordDeclTypePtr = cxxRecordDecl->getTypeForDecl();
+      clang::QualType cxxRecordTy(cxxRecordDeclTypePtr, 0);
+      clang::QualType cxxRecordPtrTy = clangCtx.getPointerType(cxxRecordTy);
+      // TODO/DOUBT: How to add `_Nonnull` as a qualifier on cxxRecordPtrTy
+      // return type ??
+
+      // Creating the type for the synthesized function
+      clang::FunctionProtoType::ExtProtoInfo EPI;
+      clang::QualType generatedFuncTy =
+          clangCtx.getFunctionType(cxxRecordPtrTy, {}, EPI);
+
+      clang::CXXMethodDecl *generatedCxxMethodDecl =
```

Comment:
The order here is wrong. In case we do not find a suitable constructor, we should not even generate a method. We should first check if there is a ctor and an operator new we can call, and only start creating methods after we are absolutely sure we will need them.